src/nbl/asset/utils/ISPIRVDebloater.cpp

```diff
@@ -0,0 +1,115 @@
+#include "nbl/asset/utils/ISPIRVDebloater.h"
+#include "nbl/asset/utils/ISPIRVOptimizer.h"
+#include "nbl_spirv_cross/spirv.hpp"
+
+#include "nbl/core/declarations.h"
+#include "nbl/system/ILogger.h"
+
+
+using namespace nbl::asset;
+
+ISPIRVDebloater::ISPIRVDebloater()
+{
+    constexpr auto optimizationPasses = std::array{
+        ISPIRVOptimizer::EOP_ELIM_DEAD_FUNCTIONS,
+        ISPIRVOptimizer::EOP_ELIM_DEAD_VARIABLES,
+        ISPIRVOptimizer::EOP_ELIM_DEAD_CONSTANTS,
+        // TODO: remove unused type. no dedicated flag for type. maybe need agressive dce to remove type.
+        ISPIRVOptimizer::EOP_TRIM_CAPABILITIES,
+    };
+    m_optimizer = core::make_smart_refctd_ptr<ISPIRVOptimizer>(std::span(optimizationPasses));
+}
+
+ISPIRVDebloater::Result ISPIRVDebloater::debloat(const  ICPUBuffer* spirvBuffer, std::span<const EntryPoint> entryPoints, system::logger_opt_ptr logger) const
+{
+    const auto* spirv = static_cast<const uint32_t*>(spirvBuffer->getPointer());
```

Comment:
basic validation missing:
- non null buffer of sufficient size to store the header and as many entry point opcodes as you have `entryPoints`
- the entry point list is not empty